### PR TITLE
Feature/reorder resonator questions

### PR DESCRIPTION
**Pull request by raverside**

### CODE REVIEW

File: src/api/routes/followers/normalizers.js [hunk 1]
```diff
@@ -47,8 +47,9 @@ export const formatLeader = (leader) => ({
  * @param {sent_resonators} sentResonator - the sent resonator for which to aggregate the questions
  * @returns {Array<Object>} - the formatted data
  */
-const aggregateResonatorQuestions = (sentResonator) =>
-    sentResonator.resonator.resonator_questions.map((question) => ({
+const aggregateResonatorQuestions = (sentResonator) => {
+    sentResonator.resonator.resonator_questions.sort((a, b) => a.order - b.order);
+    return sentResonator.resonator.resonator_questions.map((question) => ({
```
Comment (TswusayonaTsamsiyu) [inline src/api/routes/followers/normalizers.js:52]: The `sort` method also returns the sorted array, so you can chain it with the `map` and shorten the code.
Like `sentResonator.resonator.resonator_questions.sort( ... ).map( ... )`

File: src/application/resonatorStats.js [hunk 2]
```diff
@@ -26,19 +26,20 @@ export async function getResonatorStats(resonatorId) {
...
                     rank: _.get(answersMap[a.answer_id], 'rank'),
                     time: a.createdAt,
+                    order: a.order,
                     resonator: resonator.title,
                 }))
             )
             .reduce((acc, cur) => acc.concat(cur), []);
 
         const sortedAnswers = _.orderBy(answers, a => a.time, ['desc']);
+        sortedAnswers.forEach((a) => { questions.find(q => q.id === a.question_id).order = a.order });
```
Comment (TswusayonaTsamsiyu) [inline src/application/resonatorStats.js:42]: Isn't the order added in line 35 above?
Maybe the nested `map` isn't necessary?

Anyway, if you leave this, please put the function body in a new line. It's hard to read the way it is now.

File: src/domain/entities/resonator.js [hunk 3]
```diff
@@ -83,6 +84,12 @@ export default class Resonator {
         return attachment;
     }
 
+    reorderQuestion(question_id, order) {
+        const question = this.questions.find(q => q.id === question_id);
+        question.order = order;
```
Comment (TswusayonaTsamsiyu) [inline src/domain/entities/resonator.js:89]: What happens if another question already has that order value?

File: src/domain/entities/resonator.js [hunk 4]
```diff
@@ -83,6 +84,12 @@ export default class Resonator {
...
+        question.id = uuid();
```
Comment (TswusayonaTsamsiyu) [inline src/domain/entities/resonator.js:90]: Why is this needed?

**Review by TswusayonaTsamsiyu — CHANGES_REQUESTED** [review]
Mostly minor changes. Looks fine overall 😄

File: src/domain/entities/resonator.js (hunk 4 above)
Comment (raverside) [reply src/domain/entities/resonator.js:90]: If you're asking about the reorderQuestion() in general - it updates the order of an existing question (use case: Leader adds some criteria to the resonator, saves. Then comes back and reorders the questions)
if you're asking about reassigning id (line 90) - without it the record doesn't get updated and the changes are discarded. I'm not sure if there's a better way to make it update the record but I couldn't figure it out.

File: src/domain/entities/resonator.js (hunk 3 above)
Comment (raverside) [reply src/domain/entities/resonator.js:89]: It's not supposed to. All criteria are assigned a unique order value

File: src/application/resonatorStats.js (hunk 2 above)
Comment (raverside) [reply src/application/resonatorStats.js:42]: On line 35 the order is added to the [answers]. On 42 I assign to [questions]
function body on a new line - ok, makes sense.

File: src/api/routes/followers/normalizers.js (hunk 1 above)
Comment (raverside) [reply src/api/routes/followers/normalizers.js:52]: good catch, thanks, will do

File: src/application/resonators.js [hunk 5]
```diff
@@ -76,11 +76,26 @@ export async function addBulkQuestionsToResonator(resonator_id, question_ids) {
...
     await getUow().commit();
     return true;
 }
+export async function reorderQuestionsForResonator(resonator_id, criteria_order) {
+    const resonator = await resonatorRepository.findByPk(resonator_id);
+
+    if (!resonator)
+        return null;
+
+    for (const [order, question_id] of criteria_order.entries()) {
```
Comment (TswusayonaTsamsiyu) [inline src/application/resonators.js:91]: Ummm shouldn't it be `Object.entries(criteria_order)`?
or does babel somehow make this work?

**Review by TswusayonaTsamsiyu — COMMENTED** [review]
See my comment regarding something that looks like an error.
If it works then I approve

File: src/application/resonators.js (hunk 5 above)
Comment (raverside) [reply src/application/resonators.js:91]: it's actually not Object.entries() but [Array.prototype.entries()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/entries), and yeah, works fine for me

**Pull request merged**